File: DebtCountDown-jQueryMobile/src/services.android/PlanService.js
var dcd = dcd || {};
dcd.services = dcd.services || {};

function PlanService()
{
	//-------------------------------------------------------------------------
	//
	// References
	//
	//-------------------------------------------------------------------------
	
	//-----------------------------------------------------------------------------
	//
	// Methods
	//
	//-----------------------------------------------------------------------------
	
	/**
	 * 
	 */
	this.addPlan = function(plan, onResult, onFault)
	{
		$.ajax({
			url: "server/dcd_server.php",
			dataType: "json",
			type: 'POST',
			data: {
				"service":"Plan",
				"action":"addPlan",
				"name":plan.name
			},
			success: function(data, textStatus, jqXHR)
			{
				var newPlan = new PlanVO();
				newPlan.fromJSON(data);
				onResult(newPlan);
			},
			error: function(jqXHR, textStatus, errorThrown)
			{
				onFault(errorThrown);
			}
		});
	};

	/**
	 * @param plan
	 * @param onResult
	 * @param onFault
	 */
	this.deletePlan = function(plan, onResult, onFault)
	{
		$.ajax({
			url: 'server/dcd_server.php',
			dataType: "json",
			type: 'POST',
			data: {
				'service':'Plan',
				'action':'deletePlan',
				'pid':plan.pid
			},
			success: function(data, textStatus, jqXHR)
			{
				onResult(plan);
			},
			error: function(jqXHR, textStatus, errorThrown)
			{
				onFault(errorThrown);
			}
		});
	};

	/**
	 * Load all plans in the system
	 */
	this.loadAllPlans = function(onResult, onFault)
	{
		$.ajax({
			url: 'server/dcd_server.php',
			dataType: "json",
			data: {
				'service':'Plan',
				'action':'loadAllPlans'
			},
			success: function(data, textStatus, jqXHR)
			{
				var json = data;
				var plan;
				var list = new Array();

				for( var i = 0; i < json.length; i++ )
				{
					plan = new PlanVO();
					plan.fromJSON( json[i] );
					
					list.push(plan);
				}

				onResult(list);
			},
			error: function(jqXHR, textStatus, errorThrown)
			{
				onFault(errorThrown);
			}
		});
	};
	
	/**
	 * This doesn't need to be a service call because we can just pull it off 
	 * the in memory model
	 */
	this.loadPlan = function(planId, onResult, onFault)
	{
		var plan = dcd.model.planProxy.getPlanById(planId);
		
		onResult(plan);
	};
	
	/**
	 * 
	 * @param plan
	 * @param onResult
	 * @param onFault
	 */
	this.updatePlan = function (plan, onResult, onFault)
	{
		$.ajax({
			url: 'server/dcd_server.php',
			dataType: "json",
			type: 'POST',
			data: {
				'service':'Plan',
				'action':'updatePlan',
				'plan':{"name":plan.name, "pid":plan.pid}
			},
			success: function(data, textStatus, jqXHR)
			{
				onResult(plan);
			},
			error: function(jqXHR, textStatus, errorThrown)
			{
				onFault(errorThrown);
			}
		});
	};
};

dcd.services.planService = new PlanService();

File: AndroidHTML5/src/com/soatech/dcd/PlanService.java
package com.soatech.dcd;

import com.google.gson.Gson;
import android.webkit.WebView;

public class PlanService {

	//-------------------------------------------------------------------------
	//
	// Variables
	//
	//-------------------------------------------------------------------------
	
	private WebView view;
	
	private Gson gson;
	
	//-------------------------------------------------------------------------
	//
	// Constructor
	//
	//-------------------------------------------------------------------------
	
	public PlanService(WebView view) {
		this.view = view;
		
		gson = new Gson();
	}
	
	//-------------------------------------------------------------------------
	//
	// Methods
	//
	//-------------------------------------------------------------------------
	
	/**
	 * 
	 * @param plan
	 * @param onResult
	 * @param onFault
	 */
	public void addPlan(String plan, String onResult, String onFault)
	{
		PlanVO test = gson.fromJson(plan, PlanVO.class);
		test.pid = (int) Math.floor(Math.random() * 100);
		
		String url = "javascript:" + onResult + "(" + gson.toJson(test) + ");";
		view.loadUrl(url);
	}
	
	/**
	 * 
	 * @param plan
	 * @param onResult
	 * @param onFault
	 */
	public void editPlan(String plan, String onResult, String onFault)
	{
		PlanVO test = gson.fromJson(plan, PlanVO.class);
		
		String url = "javascript:" + onResult + "(" + gson.toJson(test) + ");";
		view.loadUrl(url);
	}
	
	/**
	 * 
	 * @param onResult
	 * @param onFault
	 */
	public void loadAllPlans(String onResult, String onFault) {
		String url = "javascript: " + onResult + "([{\"id\":\"1\", \"name\":\"Plan1\"}]);";
		view.loadUrl(url);
	}
	
}
